### PR TITLE
[Merged by Bors] - feat: add to_additive linter checking whether additive decl exists

### PR DESCRIPTION
**Pull request by fpvandoorn**
* Force the user to specify whether the additive declaration already exists.
* Will raise a linter error if the user specified it wrongly
* Requested on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/to_additive.20clashes)

---
neither of the following are functional dependencies, just trying to minimize merge conflicts
- [x] depends on: #1819 
- [x] depends on: #1780

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#1819
* leanprover-community/mathlib4#1780
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by fpvandoorn** [comment]
Ideally I would like to get a `bors d+` before I fix all linter errors generated by this. That allows me to go through the library only once, instead of multiple times.

File: Mathlib/Algebra/Group/Defs.lean [hunk 1]
```diff
@@ -83,17 +83,17 @@ infixl:65 " +ᵥ " => HVAdd.hVAdd
...
 infixr:73 " • " => HSMul.hSMul
 
-attribute [to_additive] Mul Div HMul instHMul HDiv instHDiv HSMul
+attribute [to_additive existing] Mul Div HMul instHMul HDiv instHDiv HSMul
 attribute [to_additive (reorder := 1) SMul] Pow
 attribute [to_additive (reorder := 1)] HPow
-attribute [to_additive (reorder := 1 5) hSMul] HPow.hPow
-attribute [to_additive (reorder := 1 4) smul] Pow.pow
+attribute [to_additive existing (reorder := 1 5) hSMul] HPow.hPow
+attribute [to_additive (reorder := 1 4) existing smul] Pow.pow
```
Comment (rwbarton) [inline Mathlib/Algebra/Group/Defs.lean:90]: should the order of these clauses be standardized (line 89 versus line 90)?

Comment (fpvandoorn) [reply Mathlib/Algebra/Group/Defs.lean:90]: oh yes, my plan was to always put it in front of the configuration. I think I was testing whether it works after as well (I don't want to enforce an order in the attribute syntax - unless you think that is better). I'll fix it when I go through the library.

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1819~~
* ~~leanprover-community/mathlib4#1780~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by kmill** [comment]
Looks like a helpful feature

bors d+

**Comment by fpvandoorn** [comment]
bors merge

**Pull request closed without merging**